packages/core-plugin-api/src/analytics/AnalyticsContext.tsx

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright 2021 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React, { createContext, ReactNode, useContext } from 'react';
+import { AnalyticsContextValue } from './types';
+
+const AnalyticsReactContext = createContext<AnalyticsContextValue>({
```

Comment:
Created a todo for the version bridge-ification.  I wonder if future context changes could warrant separate contexts.  While we are saying this is _the_ analytics context, we _actually_ just mean it's contextual information for analytics, and it happens to take the form of a context.  Future stuff might be more appropriate elsewhere anyway.

Reply:
I'd say let's rewrite it to be a version bridged context after merge but before the release, in case we want to make changes to this context in the future. It might be that there could be additional contexts too, but we don't wanna lock ourselves out of evolving this one. It also prevents this context from breaking when used across multiple versions of `core-plugin-api`